web/app/Http/Controllers/JobsController.php

```diff
@@ -17,9 +18,23 @@ class JobsController extends Controller
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-      $jobs = Jobs::orderBy('created_at', 'desc')->get();
+      $jobs = [];
+      $type = $request->get('type');
+      if($type && $type === 'need_approval') {
+          if(Auth::user()->hasUserRole && Auth::user()->hasUserRole->hasRole->name === 'moderator')
+          $jobs = Job::where('owner_id', '!=', Auth::id())->where('status', 'approval')->orderBy('created_at', 'desc')->get();          
+      } else if($type && $type === 'my_jobs') {
+          $jobs = Job::where('owner_id', Auth::id())->orderBy('created_at', 'desc')->get();
+      } else {
+          $jobs = Job::where('owner_id', '!=', Auth::id())->where('status', '!=', 'approval')->orderBy('created_at', 'desc')->get();
+      }
+
+      foreach ($jobs as $job) {
```

Comment:
You can define an accessor in the model for this. https://laravel.com/docs/5.5/eloquent-mutators#accessors-and-mutators